ContextApp/src/main/java/com/github/migi_1/ContextApp/MainActivity.java

```diff
@@ -176,37 +155,66 @@ public void onClick(View v) {
      *      The instance of this Application.
      */
     public Main getMain() {
-        
+
         return application;
-        
+
     }
 
     /**
      * Returns the application's PositionHolder.
      * @return posHolder
      */
     public PositionHolder getPosHolder() {
-        
         return posHolder;
-        
     }
-    
+
     /**
      * Returns the app's instance of the HeartsUpdateFunctions class.
      * @return huFunctions HeartsUpdateFunctions
      */
     public HeartsUpdateFunctions getHUFunctions() {
-        
         return huFunctions;
-        
     }
-    
+
     /**
      * Getter of client.
      * @return client ClientWrapper
      */
     public ClientWrapper getClient() {
-        return client;
+        return clientHub.getClientWrapper();
+    }
+
+    /**
+     * Starts the bug event. 
+     */
+    public void startBugEvent() {
+        if (!eventStarted) {
+            eventStarted = true;
+            Intent nextScreen = new Intent(getApplicationContext(), RotateBugSprayActivity.class);
+            nextScreen.putExtra("Position", position);
+            nextScreen.putExtra("BugPosition", getRandomPosition());
+            nextScreen.putExtra("SprayPosition", getRandomPosition());
+            startActivity(nextScreen);
+        }
+    }
+
+    /**
+     * Retuns a random position. 
+     * @return a random platform position.
+     */
+    private PlatformPosition getRandomPosition() {
+        int randomNumber = new Random().nextInt(4);
+        switch (randomNumber) {
+            case 0:
+                return PlatformPosition.BACKLEFT;
+            case 1:
+                return PlatformPosition.BACKRIGHT;
+            case 2:
+                return PlatformPosition.FRONTLEFT;
+            case 3:
+                return PlatformPosition.FRONTRIGHT;
+            default:
+                throw new IllegalStateException("Wrong number: " + randomNumber);
+        }
```

Comment:
instead of this switch case you can also use "return PlatformPosition.values()[randomNumber];"